worker/main.py

```
@@ -38,6 +80,48 @@ def StreamJobOutput(self, request, context):
+        with self._running_procs_lock:
+            procs = list(self._running_procs.items())
+
+        for job_id, proc in procs:
+            print(f"[worker] terminating in-flight job {job_id} for shutdown")
+            try:
+                os.killpg(proc.pid, signal.SIGTERM)
+            except ProcessLookupError:
+                # Already exited (e.g. finished between the shutdown call
+                # starting and us getting to it here) — nothing to do.
+                pass
+            except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)